src/engraving/dom/instrtemplate.cpp

```diff
@@ -853,19 +853,28 @@ const InstrumentTemplate* searchTemplateForInstrNameList(const std::list<String>
 {
     const InstrumentTemplate* bestMatch = nullptr; // default if no matches
     int bestMatchStrength = 0; // higher for better matches
-    for (const InstrumentGroup* g : instrumentGroups) {
-        for (const InstrumentTemplate* it : g->instrumentTemplates) {
-            for (const String& name : nameList) {
-                if (name.isEmpty() || it->useDrumset != useDrumset) {
+
+    for (String name : nameList) {
+        if (name.isEmpty()) {
+            continue;
+        }
+
+        if (!caseSensitive) {
+            name = name.toLower();
+        }
```

Comment:
This might help too by avoiding copying strings unnecessarily:

```C++
for (const String& name : nameList) {
    if (name.isEmpty()) {
        continue;
    }

    const String& normalizedName = caseSensitive ? name : name.toLower();
    StaffName instrName(normalizedName);
}
```